python/fast_mlsirm/multilevel/estimation.py

```
@@ -127,4 +197,400 @@ def weighted_contextual_effect(
+        raise ValueError("responses must be a NumPy ndarray")
+    if responses.ndim != 2:
+        raise ValueError("responses must be a two-dimensional occasion-by-item matrix")
+    if responses.shape[0] != n_occasions:
+        raise ValueError("responses rows must align with the sealed occasion order")
+    if responses.shape[1] < 2:
+        raise ValueError("hierarchical CT-AR Rasch requires at least two items")
+    if responses.dtype == np.bool_ or np.issubdtype(responses.dtype, np.bool_):
+        raise ValueError("responses must be 0, 1, or NaN rather than Boolean values")
+    try:
+        matrix = np.ascontiguousarray(responses, dtype=np.float64)
+    except Exception:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)